mediadevices.go

```diff
@@ -217,8 +217,8 @@ func selectBestDriver(filter driver.FilterFn, constraints MediaTrackConstraints)
 		return nil, MediaTrackConstraints{}, errNotFound
 	}
 
-	constraints.selectedMedia = bestProp
-	constraints.selectedMedia.Merge(constraints.MediaConstraints)
+	constraints.selectedMedia.MergeConstraints(constraints.MediaConstraints)
```

Comment:
It would be safer to clear `constraints.selectedMedia` by `prop.Media{}` before merge.

Reply:
Yep, that makes sense.